rclcpp_examples/src/topics/talker.cpp

```diff
@@ -13,13 +13,28 @@
 // limitations under the License.
 
 #include <iostream>
+#include <string>
 
 #include "rclcpp/rclcpp.hpp"
 
 #include "std_msgs/msg/string.hpp"
 
+#define STRING_(s) #s
+#define STRING(s) STRING_(s)
+
 int main(int argc, char * argv[])
 {
+  #ifdef EXPECTED_RMW_IMPLEMENTATION_IDENTIFIER
+  std::string rmw_identifier(rmw_get_implementation_identifier());
+
+  if (rmw_identifier.compare(STRING(EXPECTED_RMW_IMPLEMENTATION_IDENTIFIER)) != 0) {
+    std::cerr << "Expected RMW implementation identifier of '" <<
+      STRING(EXPECTED_RMW_IMPLEMENTATION_IDENTIFIER) << "' but instead found '" <<
+      rmw_identifier << "'" << std::endl;
+    return 1;
+  }
+  #endif
```

Comment:
This is the kind of thing that I didn't want to have in the examples. (I think it detracts from the otherwise simple example code)

I was thinking we could instead put the ability to check this into the `rmw` library itself. I never prototyped this, but I'm still interested in doing it.

If we merge this as-is, I'd want a TODO to remind us that there is probably a way to do it which doesn't affect the user code here.